src/shell/filesystem_shell.rs

```diff
@@ -95,132 +96,46 @@ impl Shell for FilesystemShell {
-                let entries = std::fs::read_dir(&entry);
-                let entries = match entries {
-                    Err(e) => {
-                        if let Some(s) = path {
-                            return Err(ShellError::labeled_error(
-                                e.to_string(),
-                                e.to_string(),
-                                s.tag(),
-                            ));
-                        } else {
-                            return Err(ShellError::labeled_error(
-                                e.to_string(),
-                                e.to_string(),
-                                name_tag,
-                            ));
-                        }
-                    }
-                    Ok(o) => o,
-                };
-                let mut entries = entries.collect::<Vec<Result<std::fs::DirEntry, _>>>();
-                entries.sort_by(|x, y| match (x, y) {
-                    (Ok(entry1), Ok(entry2)) => entry1
-                        .path()
-                        .to_string_lossy()
-                        .to_lowercase()
-                        .cmp(&entry2.path().to_string_lossy().to_lowercase()),
-                    (Err(_), Ok(_)) => std::cmp::Ordering::Greater,
-                    (Ok(_), Err(_)) => std::cmp::Ordering::Greater,
-                    _ => std::cmp::Ordering::Equal,
-                });
-                let stream = async_stream! {
-                    for entry in entries {
-                        if ctrl_c.load(Ordering::SeqCst) {
-                            break;
-                        }
-                        if let Ok(entry) = entry {
-                            let filepath = entry.path();
-                            if let Ok(metadata) = std::fs::symlink_metadata(&filepath) {
-                                let mut filename = if let Ok(fname) = filepath.strip_prefix(&cwd) {
-                                    fname
-                                } else {
-                                    Path::new(&filepath)
-                                };
-
-                                if short_names {
-                                    filename = if let Some(fname) = filename.file_name() {
-                                        Path::new(fname)
-                                    } else {
-                                        let fname = filename
-                                            .file_name()
-                                            .or_else(|| Path::new(filename).file_name())
-                                            .unwrap_or(filename.as_os_str());
-                                        Path::new(fname)
-                                    }
-                                }
+        let paths: Vec<GlobResult> = match glob::glob(&path.to_string_lossy()) {
+            Ok(g) => Ok(g),
+            Err(e) => Err(ShellError::labeled_error("Glob error", e.msg, &p_tag)),
```

Comment:
If I'm in a directory that has no files, and go `ls`, will I see this error?

Reply:
Yes it would.... Not sure how to handle this....

Reply:
Yeah, this is a tough one. Nothing comes to mind at first, so let's both keep thinking, since this would be undesirable behaviour.